### PR TITLE
Add ::dotWrite method

### PR DESCRIPTION
**Pull request by jodiedunlop**
Added new method for mutating arrays with depth based keys:
`ArrayUtility::dotWrite($array, $key, $value)`

Added new method for testing existence even when the value is null:
`ArrayUtility::dotExists($array, $key)`

Included:
- Tests
- Updated README

### CODE REVIEW

File: src/ArrayUtility.php [hunk 1]
```diff
@@ -119,13 +119,72 @@ public static function dotRead(array $array = null, $key, $default_value = null)
...
 	}
 
+    /**
+     * Write a value to a nested array
+     * @param array  $array
+     * @param string $writeKey The write key e.g. "level1.level2.key"
+     * @param mixed  $value    Value to write
+     * @return bool  Returns true if write was successful
+     */
+    public static function dotWrite(array &$array, $writeKey, $value = null)
```
Comment (ababkov) [inline src/ArrayUtility.php]: Should be immuteable - how come array ref is getting passed in?

File: src/ArrayUtility.php [hunk 2]
```diff
@@ -119,13 +119,72 @@ public static function dotRead(array $array = null, $key, $default_value = null)
...
+
+            // Last key
+            $ref[$key] = $value;
+        }
+
+        return true;
+    }
+
+    /**
+     * Read a value from a nested array using a single string
```
Comment (ababkov) [inline src/ArrayUtility.php]: Probably needs to be updated.

**Review by ababkov — CHANGES_REQUESTED** [review]

File: src/ArrayUtility.php (hunk 1 above)
Comment (jodiedunlop) [reply src/ArrayUtility.php]: Should it be immutable? Felt like I wanted to mutate the array without causing a copy op.

Comment (ababkov) [reply src/ArrayUtility.php]: All the others are immutable

Comment (jodiedunlop) [reply src/ArrayUtility.php]: Added ::dotMutate, and changed ::dotWrite

File: src/ArrayUtility.php (hunk 2 above)
Comment (jodiedunlop) [reply src/ArrayUtility.php]: Updated

**Review by ababkov — APPROVED** [review]

**Pull request merged**